oct/Makefile.in

```diff
@@ -1040,54 +1040,58 @@ minigzip.c:
 	https://raw.githubusercontent.com/madler/zlib/master/test/$@
 
 .PRECIOUS: %.uncompressed
-empty.uncompressed:
+$(srcdir)/empty.uncompressed:
 	cd $(srcdir) && dd if=/dev/null bs=1k count=1 of=$@
 
-random4k.uncompressed:
+$(srcdir)/random4k.uncompressed:
 	cd $(srcdir) && dd if=/dev/urandom bs=4k count=1 of=$@
 
-random13M.uncompressed:
+$(srcdir)/random13M.uncompressed:
 	cd $(srcdir) && dd if=/dev/urandom bs=1M count=13 of=$@
 
-sparse10M.uncompressed:
+$(srcdir)/sparse10M.uncompressed:
 	cd $(srcdir) && dd if=/dev/urandom bs=1 count=0 seek=10M of=$@
 
-sparse1000M.uncompressed:
+$(srcdir)/sparse1000M.uncompressed:
 	cd $(srcdir) && dd if=/dev/urandom bs=1 count=0 seek=1000M of=$@
 
-zero4k.uncompressed:
+$(srcdir)/zero4k.uncompressed:
 	cd $(srcdir) && dd if=/dev/zero bs=4k count=1 of=$@
 
-zero13M.uncompressed:
+$(srcdir)/zero13M.uncompressed:
 	cd $(srcdir) && dd if=/dev/zero bs=1M count=13 of=$@
 
-%.uncompressed: %.source download.sh config.sh
+# This rule requires config.sh an download.sh, but we don't want to re-download
+# all the files every time config.sh is regenerated or download.sh is modified.
+# We just assume the user executed configure before running this and that
+# download.sh is always available.
+%.uncompressed: %.source
```

Comment:
Another alternative here is to use an [order-only prerequisites](https://www.gnu.org/software/make/manual/make.html#Prerequisite-Types), so it doesn't check `config.sh` and `donwload.sh`'s mtime:
```
%.uncompressed: %.source | download.sh config.sh
```

Reply:
This is much better!
Thanks!
Fixed.